src/main/java/org/opensearch/knn/index/vectorvalues/VectorValueExtractorStrategy.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.knn.index.vectorvalues;
+
+import org.apache.lucene.index.BinaryDocValues;
+import org.apache.lucene.index.ByteVectorValues;
+import org.apache.lucene.index.FloatVectorValues;
+import org.apache.lucene.search.DocIdSetIterator;
+import org.apache.lucene.util.ArrayUtil;
+import org.apache.lucene.util.BytesRef;
+import org.opensearch.knn.index.VectorDataType;
+import org.opensearch.knn.index.codec.util.KNNVectorSerializer;
+import org.opensearch.knn.index.codec.util.KNNVectorSerializerFactory;
+
+import java.io.IOException;
+
+/**
+ * Provides different strategies to extract the vectors from different {@link KNNVectorValuesIterator}
+ */
+interface VectorValueExtractorStrategy {
+    /**
+     * Extract Vector based on the vector datatype and vector values iterator.
+     * @param vectorDataType {@link VectorDataType}
+     * @param vectorValuesIterator {@link KNNVectorValuesIterator}
+     * @return vector
+     * @param <T> could be of type float[], byte[]
+     * @throws IOException exception during extracting the vectors
+     */
+    <T> T extract(VectorDataType vectorDataType, KNNVectorValuesIterator vectorValuesIterator) throws IOException;
+
+    /**
+     * Strategy to extract the vector from {@link KNNVectorValuesIterator.DocIdsIteratorValues}
+     */
+    class DISIVectorExtractor implements VectorValueExtractorStrategy {
+        @Override
+        public <T> T extract(final VectorDataType vectorDataType, final KNNVectorValuesIterator vectorValuesIterator) throws IOException {
+            final DocIdSetIterator docIdSetIterator = vectorValuesIterator.getDocIdSetIterator();
+            switch (vectorDataType) {
+                case FLOAT:
+                    if (docIdSetIterator instanceof BinaryDocValues) {
+                        final BinaryDocValues values = (BinaryDocValues) docIdSetIterator;
+                        return (T) getFloatVectorFromByteRef(values.binaryValue());
```

Comment:
Can we write a unit test calling this code part? This code part might throw `ClassCastException`

Reply:
this is already covered by the UTs present here: KNNVectorValuesTests. If you see I have added tests for types of vectors, byte[], binary and float[]. so this casting case is covered.

Reply:
I think I got it what you mean.  I will add a test for exception.

Reply:
My bad. Never mind. I thought (T) casting is not allowed but it was not..

Reply:
I thought you were asking for the last error condition. So I added tests for that.